engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -37,16 +38,21 @@ object typing {
   object TypedObjectTypingResult {
 
     def apply(definition: TypedObjectDefinition): TypedObjectTypingResult =
-    //we don't use mapValues here to avoid lazy evaluation that crashes during serialization...
       TypedObjectTypingResult(definition.fields.map { case (k, v) => (k, Typed(v))})
 
-    def apply(fields: Map[String, TypingResult]): TypedObjectTypingResult =
-      TypedObjectTypingResult(fields, TypedClass(classOf[java.util.Map[_, _]], List(Typed[String], Unknown)))
+    def apply(fields: List[(String, TypingResult)]): TypedObjectTypingResult =
+      TypedObjectTypingResult(ListMap(fields: _*))
+
+    def apply(fields: List[(String, TypingResult)], objType: TypedClass): TypedObjectTypingResult =
+      TypedObjectTypingResult(ListMap(fields: _*), objType)
 
+    def apply(fields: ListMap[String, TypingResult]): TypedObjectTypingResult =
+      TypedObjectTypingResult(fields, TypedClass(classOf[java.util.Map[_, _]], List(Typed[String], Unknown)))
   }
 
-  case class TypedObjectTypingResult(fields: Map[String, TypingResult],
-                                     objType: TypedClass, additionalInfo: Map[String, AdditionalDataValue] = Map.empty) extends SingleTypingResult {
+  case class TypedObjectTypingResult(fields: ListMap[String, TypingResult],
```

Comment:
1. Please left the comment like:
`Warning, fields are kept in list-like map so: 1) order is important 2) lookup has O(n) complexity`
2. Please add notice in `MigrationGuide.md` about this change

Reply:
:+1:

Reply:
https://github.com/TouK/nussknacker/pull/1532/commits/49e9f129a8634a4ee724d7774e7b254480bd48f7